test/testPriceRouter/PriceRouter.t.sol

```diff
@@ -0,0 +1,788 @@
+                    address(USDC),
+                    price,
+                    badMinPrice
+                )
+            )
+        );
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(stor), 1e8);
+    }
+
+    /**
+     * @notice All pricing operations go through `_getValueInUSD`, so checking for revert in `addAsset` is sufficient.
+     */
+    function testAssetAboveMaxPrice() external {
+        // Store price of USDC.
+        uint256 price = uint256(IChainlinkAggregator(USDC_USD_FEED).latestAnswer());
+
+        // Add USDC again, but set a bad maxPrice.
+        uint144 badMaxPrice = 0.9e8;
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, USDC_USD_FEED);
+        PriceRouter.ChainlinkDerivativeStorage memory stor = PriceRouter.ChainlinkDerivativeStorage(
+            badMaxPrice,
+            0,
+            0,
+            false
+        );
+
+        // Simulate calling startEditAsset.
+        bytes32 editHash = keccak256(abi.encode(USDC, settings, abi.encode(stor)));
+        stdstore
+            .target(address(priceRouter))
+            .sig(priceRouter.assetEditableTimestamp.selector)
+            .with_key(editHash)
+            .checked_write(uint256(1));
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__AssetAboveMaxPrice.selector,
+                    address(USDC),
+                    price,
+                    badMaxPrice
+                )
+            )
+        );
+        priceRouter.completeEditAsset(USDC, settings, abi.encode(stor), 1e8);
+    }
+
+    function testAssetStalePrice() external {
+        // Store timestamp of USDC.
+        uint256 timestamp = uint256(IChainlinkAggregator(USDC_USD_FEED).latestTimestamp());
+        timestamp = block.timestamp - timestamp;
+
+        // Advance time so that the price becomes stale.
+        vm.warp(block.timestamp + 1 days);
+
+        vm.expectRevert(
+            bytes(
+                abi.encodeWithSelector(
+                    PriceRouter.PriceRouter__StalePrice.selector,
+                    address(USDC),
+                    timestamp + 1 days,
+                    1 days
+                )
+            )
+        );
+        priceRouter.getValue(USDC, 1e6, USDC);
+    }
+
+    function testETHtoUSDPriceFeedIsChecked() external {
+        // Check if querying an asset that needs the ETH to USD price feed, that the feed is checked.
+        // Add BOND as an asset.
+        PriceRouter.AssetSettings memory settings = PriceRouter.AssetSettings(CHAINLINK_DERIVATIVE, BOND_ETH_FEED);
+        PriceRouter.ChainlinkDerivativeStorage memory stor = PriceRouter.ChainlinkDerivativeStorage(
+            100e18,
+            0.0001e18,
+            2 days,
+            true
+        );
+
+        // Simulate calling startEditAsset.
+        bytes32 editHash = keccak256(abi.encode(BOND, settings, abi.encode(stor)));
```

Comment:
Why are we simulating `startEditAsset()` throughout the tests vs just using it? Arguably, there should be tests that implement this as that is the true happy path vs using storage manipulation with foundry right?

Reply:
Because if we do the real deal, then the test needs to have its block timestamp fast forwarded by 7 days, which causes all the chainlink price feed checks to revert bc most of them use a 1 day staleness check. We could get away with doing a true happy path test, if we created mock price feeds, but given the audit is basically starting, I dont think it is worth the extra time.

Reply:
It sounds like moving forward with the mock set up for a true happy path test in the future (aka closer to when BPT cellars or other cellars requiring `PriceRouterV2` is needed) would be good. Let's open an issue on this and address later.